### ISSUE TITLE
IDN Conversion Failed (no errors)

### ISSUE THREAD

**Issue by theteknocat** [guzzle/guzzle#2636]
**Guzzle version(s) affected**: 6.5.3
**PHP version**: 7.2.30
**cURL version**: 7.19.7

**Description**  
I am using Laravel, and the Bestit Harvest API library which uses Guzzle to make Harvest API requests. I just deployed the latest updates to our production server, and now I receive this error. I'm guessing it must be doing the first call to idn_to_ascii on line 35 of Utils.php, since there is no $info array with an 'errors' property present so it doesn't show any errors in brackets after the error message.

I do not have this issue on my local machine, on which I'm running PHP 7.2.29 and cURL 7.69.1.

**How to reproduce**  
I expect this to be rather challenging to reproduce without the exact server environment, given it only occurs in our particular environment. I'm at a loss on how to troubleshoot and reproduce this, hence I'm reporting this as an issue.

**Possible Solution**  
I'm expecting the solution to be somewhere in PHP configuration or something, not in the Guzzle codebase, given it only seems to happen in a certain server environment.

I'm hoping somebody may have some suggestions about what aspects of PHP configuration could affect this.

**Comment by theteknocat** [comment]
Oh, I should specify that I'm running Laravel 6.18.14.

**Comment by theteknocat** [comment]
I made a bit of progress today - not sure what I did, but whatever it was got the server to give me a more specific error.

I'm now getting: idn_to_ascii(): INTL_IDNA_VARIANT_2003 is deprecated

This is therefore definitely some sort of server configuration issue, so I can follow up with our hosting service as they provide managed support.

**Comment by GrahamCampbell** [comment]
Please check existing bug reports before posting. There is already an issue for this #2620 and a fix: #2620.

**Comment by theteknocat** [comment]
I think this issue can be closed as I figured out that the problem for me was nothing to do with the Guzzle code but because the versions of PHP on our server had an intl extension that was compiled against an older version of ICU.

I also discovered in digging through the code in Laravel that Symfony implements it's own polyfils for this functionality if the intl extension does not exist in PHP. As such, my simple solution was to just disable the intl extension altogether on the server in the PHP configuration.

I hope this will at least provide some other people out there having the same issue with a possible solution.
